design/pool-configuration.md

```diff
@@ -0,0 +1,553 @@
+* [PR: Limit announcement of address pools to specific BGP peers](https://github.com/metallb/metallb/pull/596)
+
+Given the IP, here we want to choose what BGP peers it must be announced to:
+
+[<img src="./images/pool_peer_selector.svg" width="400"/>](./images/pool_peer_selector.svg)
+
+On top of that, different nodes must announce the IP to a different set of BGP peers:
+
+[<img src="./images/pool_peer_node_selector.svg" width="400"/>](./images/pool_peer_node_selector.svg)
+
+#### No announcement
+
+This a degenerate case that was suggested by [this discussion](https://github.com/metallb/metallb/pull/832#issuecomment-888897400) where
+there are scenarios in which only the IP pooling side of metallb is used, but the routes to those IPs are set statically.
+
+On top of that, Calico suggests to deploy MetalLB without the speakers to leverage only the allocation feature exposed by the controller in
+[its official documentation](https://docs.projectcalico.org/networking/advertise-service-ips).
+
+### Using a given AddressPool only for a subset of services
+
+Currently, the options to make sure a given service is using a specific IP from a specific pool are:
+
+* using the `metallb.universe.tf/address-pool` annotation on the service
+* using the `auto-assign: false` parameter on the pool, and then statically assign the IP to a given service.
+
+From this point of view, we can enhance the allocation logic allowing users to reserve a given pool for a set of namespaces or for a set
+of services. This is more flexible and moves the focus on the cluster administrator, who is setting the MetalLB configuration
+in a multi tenant scenario.
+
+This, in combination with the other configuration degrees above, will provide a way to have per service (or per namespace) behavior
+ (i.e. specifying the communities for a given service).
+
+The issues related to this arguments:
+
+* [Issue: Pool affinity for Namespace](https://github.com/metallb/metallb/issues/383)
+* [Issue: Feature request: addresses assignment outside of Service definition](https://github.com/metallb/metallb/issues/616)
+* [Issue: allow selecting services to announce with the bgb speaker](https://github.com/metallb/metallb/issues/604)
+* [Issue: Dynamic BGP communities from services](https://github.com/metallb/metallb/issues/934)
+
+### Risks and Mitigations
+
+#### Regressions
+
+The e2e test suite is wide enough to ensure we have good coverage on metallb's main features. Given the new data model will be functionally equivalent
+to the old one, converting the test suite should give us enough confidence in that regard.
+
+#### Impact on the users' configurations
+
+Deprecating the configmap will have impact on the users' current configurations. To mitigate this, we may consider having a controller that converts the configmap
+into the CRDs, or provide some offline tool in charge of that (the latter option is more desirable, as it will remove the need of reconciling changes into the configmap).
```

Comment:
I guess this implies for at least for 1 release it will be possible to use CRDs or a configmap to configure metalLB (but not both), right?

Reply:
An automatic CLI converter could also help and possibly ease the transition, that's another option.

Reply:
Thanks!